test/libsolidity/syntaxTests/literals/ternary_operator_return_type.sol

```diff
@@ -0,0 +1,19 @@
+contract TestTernary
+{
+    mapping(uint8 => bool) flags;
+
+    function g() public returns (uint256 result)
+    {
+        flags[uint8(1)] = true;
+        flags[uint8(2)] = true;
+        flags[uint8(3)] = true;
+
+        result = (flags[0] ? uint256(100) : 0) +
+                    (flags[1] ? 110 : 0) +
+                    (flags[2] ? 125 : 0) +
+                    (flags[3] ? 150 : 0);
```

Comment:
Let's include more cases None of them should produce a warning now but if we introduce it later, we'll want to see which cases it affects.

```solidity
        bool t = true;
        bool f = false;
        uint8 v255 = 255;
        uint8 v63 = 63;
        uint8 a;

        // Currently none of these should produce errors or warnings.
        // The result of the operator is always a limited-precision integer, even if all arguments are literals.

        a = (t ? 63 : 255) + (f ? 63 : 255);
        a = (t ? 0x3f : 0xff) + (f ? 0x3f : 0xff);
        a = (t ? uint8(63) : 255) + (f ? 63 : uint8(255));
        a = (t ? v63 : 255) + (f ? 63 : v255);

        a = (true ? 63 : 255) + (false ? 63 : 255);
        a = (true ? 0x3f : 0xff) + (false ? 0x3f : 0xff);
        a = (true ? uint8(63) : 255) + (false ? 63 : uint8(255));
        a = (true ? v63 : 255) + (false ? 63 : v255);

        a = (t ? 63 : 255) - (f ? 63 : 255);
        a = (t ? 63 : 255) * (f ? 63 : 255);
        a = (t ? 63 : 255) / (f ? 63 : 255);

        a = (t ? (true ? 63 : 255) : (false ? 63 : 255)) + (f ? (t ? 63 : 255) : (f ? 63 : 255));
        a = uint8(t ? 63 : 255) + uint8(f ? 63 : 255);
```

Also, let's rename the test to `ternary_operator_return_type_with_literal_arguments.sol`